src/mbgl/geometry/dem_data.cpp

```diff
@@ -13,13 +13,22 @@ DEMData::DEMData(const PremultipliedImage& _image):
         throw std::runtime_error("raster-dem tiles must be square.");
     }
 
+    auto decodeRGB = [&] (const uint8_t r, const uint8_t g, const uint8_t b) {
+        if (encoding == Tileset::Encoding::Mapbox) {
```

Comment:
Instead of using an if, as it's called inside a loop, you could change that to a Functor so it gets inlined and resembles what you did in the js version, selecting the decode function before the loop.